dali/pipeline/data/tensor_list.h

```diff
@@ -603,6 +609,62 @@ class DLL_PUBLIC TensorList : public Buffer<Backend> {
   std::list<Tensor<Backend> > tensor_views_;
 
   USE_BUFFER_MEMBERS();
+
+ private:
+  /** @defgroup ContiguousAccessorMembers Contiguous accessor member functions
+   * The Buffer-like access to TensorList is now removed, and sample-by-sample approach
+   * should be used. We hide those functions as private, but leave it for the TensorList
+   * implementation to use.
+   * @{
+   */
+
+  template <typename T>
+  inline T* mutable_data() {
+    return Buffer<Backend>::template mutable_data<T>();
+  }
+
+  template <typename T>
+  inline const T* data() const {
+    return Buffer<Backend>::template data<T>();
+  }
+
+  inline void* raw_mutable_data() {
+    return Buffer<Backend>::raw_mutable_data();
+  }
+
+  inline const void* raw_data() const {
+    return Buffer<Backend>::raw_data();
+  }
+
+  /** @} */  // end of ContiguousAccessorMembers
+
+
+  /** @defgroup ContiguousAccessorFunctions Fallback contiguous accessors
+   * Fallback access to contiguous data to TensorList. It should not be used for processing,
+   * and can be used only for outputs of the pipeline that were made sure to be contiguous.
+   * Currently TensorList is contiguous by design, but it is up to change.
+   * @{
+   */
+
+  /**
+   * @brief Return an un-typed pointer to the underlying storage.
+   * Buffer must be either empty or have a valid type and be contiguous.
+   */
+  friend void *unsafe_raw_mutable_data(TensorList<Backend> &tl) {
```

Comment:
Why we need that function? To highlight that this approach is going to be removed?

Reply:
So, I think we are going to keep some sort of this behaviour for pipeline output. But I don't want anyone using TensorList.raw_data() & co inside of regular code base (that is operator implementation etc). So this is the escape hatch that I mention in the PR description, but it's also documented and named with the `unsafe_` prefix to discourage the usage. The "I know what I'm doing" kind of code.

Reply:
Usually it is rather `I do whatever possible to make that working`.